heudiconv/cli/run.py

```diff
@@ -140,238 +30,147 @@ def get_parser():
-    # pre-process provided list of files and possibly sort into groups/sessions
-    # Group files per each study/sid/session
-
-    outdir = op.abspath(args.outdir)
-
-    try:
-        import etelemetry
-        latest = etelemetry.get_project("nipy/heudiconv")
-    except Exception as e:
-        lgr.warning("Could not check for version updates: %s", str(e))
-        latest = {"version": 'Unknown'}
-
-    lgr.info(INIT_MSG(packname=__packagename__,
-                      version=__version__,
-                      latest=latest["version"]))
-
-    if args.command:
-        process_extra_commands(outdir, args)
-        return
-    #
-    # Load heuristic -- better do it asap to make sure it loads correctly
-    #
-    if not args.heuristic:
-        raise RuntimeError("No heuristic specified - add to arguments and rerun")
-
-    if args.queue:
-        lgr.info("Queuing %s conversion", args.queue)
-        iterarg, iterables = ("files", len(args.files)) if args.files else \
-                             ("subjects", len(args.subjs))
-        queue_conversion(args.queue, iterarg, iterables, args.queue_args)
-        sys.exit(0)
-
-    heuristic = load_heuristic(args.heuristic)
-
-    study_sessions = get_study_sessions(args.dicom_dir_template, args.files,
-                                        heuristic, outdir, args.session,
-                                        args.subjs, grouping=args.grouping)
-
-    # extract tarballs, and replace their entries with expanded lists of files
-    # TODO: we might need to sort so sessions are ordered???
-    lgr.info("Need to process %d study sessions", len(study_sessions))
-
-    # processed_studydirs = set()
-
-    for (locator, session, sid), files_or_seqinfo in study_sessions.items():
-
-        # Allow for session to be overloaded from command line
-        if args.session is not None:
-            session = args.session
-        if args.locator is not None:
-            locator = args.locator
-        if not len(files_or_seqinfo):
-            raise ValueError("nothing to process?")
-        # that is how life is ATM :-/ since we don't do sorting if subj
-        # template is provided
-        if isinstance(files_or_seqinfo, dict):
-            assert(isinstance(list(files_or_seqinfo.keys())[0], SeqInfo))
-            dicoms = None
-            seqinfo = files_or_seqinfo
-        else:
-            dicoms = files_or_seqinfo
-            seqinfo = None
-
-        if locator == 'unknown':
-            lgr.warning("Skipping unknown locator dataset")
-            continue
-
-        anon_sid = anonymize_sid(sid, args.anon_cmd) if args.anon_cmd else None
-        if args.anon_cmd:
-            lgr.info('Anonymized {} to {}'.format(sid, anon_sid))
-
-        study_outdir = op.join(outdir, locator or '')
-        anon_outdir = args.conv_outdir or outdir
-        anon_study_outdir = op.join(anon_outdir, locator or '')
-
-        # TODO: --datalad  cmdline option, which would take care about initiating
```

Comment:
There is a datalad cmdline option now. Can this be removed?

Reply:
I guess -- good find